### PR TITLE
AF-1865: [Streamline Development Lifcycle] Fix backguard compatibilities and improve 'dev mode' deployment policy

### PR DESCRIPTION
**Pull request by pefernan**
This change solves some Kie Server backwards compatibility issues, the changes are:
- DEVELOPMENT mode will allow deploy and update any type of modules, not only SNAPSHOTs. DEVELOPMENT mode will still be the default mode.
- Renamed REGULAR mode into PRODUCTION mode. It will keep the same Kie Server deployment policy.

@tomasdavidorg @MarianMacik @mswiderski could you please take a look?

This is part of an ensemble, please merge with:
https://github.com/kiegroup/droolsjbpm-integration/pull/1747
https://github.com/kiegroup/kie-wb-common/pull/2497
https://github.com/kiegroup/kie-docs/pull/1409

### CODE REVIEW

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ public KieServerImpl(KieServerStateRepository stateRepository) {
             logger.info("Starting server in '" + mode.name() + "' mode.");
         } catch (Exception ex) {
             mode = KieServerMode.DEVELOPMENT;
-            logger.warn("Unable to parse value of " + KieServerConstants.KIE_SERVER_MODE + " = " + modeParam + "; supported values are 'REGULAR' or 'DEVELOPMENT'. Falling back to 'DEVELOPMENT' mode.");
+            logger.warn("Unable to parse value of " + KieServerConstants.KIE_SERVER_MODE + " = " + modeParam + "; supported values are 'REGULAR' or 'PRODUCTION'. Falling back to 'DEVELOPMENT' mode.");
```
Comment (porcelli) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java]: REGULAR' or 'PRODUCTION?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/AbstractKieServerImplTest.java [hunk 2]
```diff
@@ -130,7 +125,7 @@ public void setupKieServerImpl() throws Exception {
         kieServer.init();
     }
 
-    private String getVersion(KieServerMode mode) {
+    protected String getVersion(KieServerMode mode) {
         return mode.equals(KieServerMode.DEVELOPMENT) ? DEVELOPMENT_MODE_VERSION : REGULAR_MODE_VERSION;
```
Comment (porcelli) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/AbstractKieServerImplTest.java]: Regular?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/AbstractKieServerImplTest.java [hunk 3]
```diff
@@ -694,8 +647,8 @@ protected void createEmptyKjar(String artifactId, String version) {
         KieServices kieServices = KieServices.Factory.get();
         KieFileSystem kfs = kieServices.newKieFileSystem();
         releaseId = kieServices.newReleaseId(GROUP_ID, artifactId, version);
-        KieModule kieModule = kieServices.newKieBuilder(kfs ).buildAll().getKieModule();
-        KieMavenRepository.getKieMavenRepository().installArtifact(releaseId, (InternalKieModule)kieModule, createPomFile(artifactId, version ) );
+        KieModule kieModule = kieServices.newKieBuilder(kfs).buildAll().getKieModule();
```
Comment (porcelli) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/AbstractKieServerImplTest.java]: if this is just format, please revert, so make easier to review

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java [hunk 4]
```diff
@@ -457,26 +451,26 @@ public void createContainer(String id, KieContainerInstance kieContainerInstance
...
             logger.debug("Container {} created successfully by extension {}", id, this);
         } catch (Exception e) {
-            Throwable root = ExceptionUtils.getRootCause( e );
-            if ( root == null ) {
+            Throwable root = ExceptionUtils.getRootCause(e);
+            if (root == null) {
                 root = e;
             }
-            messages.add(new Message(Severity.ERROR, "Error when creating container " + id +" by extension " + this + " due to " + root.getMessage()));
+            messages.add(new Message(Severity.ERROR, "Error when creating container " + id + " by extension " + this + " due to " + root.getMessage()));
```
Comment (porcelli) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/JbpmKieServerExtension.java]: lots of formating changes, please try avoid to make the review easier

**Comment by pefernan** [comment]
@porcelli you're right I always use to format all files. Better now?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 1 above)
Comment (porcelli) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java]: @pefernan here the message should be:

logger.warn("Unable to parse value of " + KieServerConstants.KIE_SERVER_MODE + " = " + modeParam + "; supported values are 'DEVELOPMENT' or 'PRODUCTION'. Falling back to 'DEVELOPMENT' mode.");

Comment (pefernan) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java]: @porcelli I missed this!

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
Jenkins retest this please

**Comment by pefernan** [comment]
Jenkins retest this please

**Comment by pefernan** [comment]
Jenkins retest this please

**Comment by ederign** [comment]
Jenkins retest this please.

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
@porcelli @ederign greens...

**Pull request merged**